lib/ReactViews/Custom/Chart/BottomDockChart.tsx

```diff
@@ -0,0 +1,367 @@
+      >
+        <Legends width={plotWidth} chartItems={processedChartItems} />
+        <div style={{ position: "relative" }}>
+          <svg
+            width="100%"
+            height={height}
+            onMouseMove={setMouseCoordsFromEvent}
+            onMouseLeave={() => setMouseCoords(undefined)}
+          >
+            <Group left={adjustedMargin.left} top={adjustedMargin.top}>
+              <RectClipPath
+                id="plotClip"
+                width={plotWidth}
+                height={plotHeight}
+              />
+              <XAxis
+                top={plotHeight + 1}
+                scale={xScale}
+                label={xAxis.units || (xAxis.scale === "time" ? "Date" : "")}
+              />
+              {yAxes.map((y, i) => (
+                <YAxis
+                  {...y}
+                  key={`y-axis-${y.units}`}
+                  color={yAxes.length > 1 ? y.color : DEFAULT_GRID_COLOR}
+                  offset={i * 50}
+                />
+              ))}
+              {yAxes.map((y) => (
+                <GridRows
+                  key={`grid-${y.units}`}
+                  width={plotWidth}
+                  height={plotHeight}
+                  scale={y.scale}
+                  numTicks={Y_AXIS_NUM_TICKS}
+                  stroke={yAxes.length > 1 ? y.color : DEFAULT_GRID_COLOR}
+                  lineStyle={{ opacity: 0.3 }}
+                />
+              ))}
+              <svg
+                id="zoomSurface"
+                clipPath="url(#plotClip)"
+                pointerEvents="all"
+              >
+                <rect
+                  width={plotWidth}
+                  height={plotHeight}
+                  fill="transparent"
+                />
+                {cursorX && <Cursor x={cursorX} stroke={DEFAULT_GRID_COLOR} />}
+                <Plot
+                  chartItems={processedChartItems}
+                  initialScales={initialScales}
+                  zoomedScales={zoomedScales}
+                />
+              </svg>
+            </Group>
+          </svg>
+          <Tooltip {...tooltip} />
+          <PointsOnMap chartItems={processedChartItems} />
+        </div>
+      </ZoomX>
+    );
+  }
+);
+
+Chart.displayName = "Chart";
+
+// Type guard to filter ChartItems that don't produce a nearestPoint
+const pointNotUndefined = (itemPoint: {
+  chartItem: ChartItem;
+  point?: ChartPoint;
+}): itemPoint is { chartItem: ChartItem; point: ChartPoint } =>
+  itemPoint.point !== undefined;
+
+/**
+ * Sorts chartItems so that `momentPoints` are rendered on top then
+ * `momentLines` and then any other types.
+ * @param {ChartItem[]} chartItems array of chartItems to sort
+ */
+const sortChartItemsByType = (chartItems: readonly ChartItem[]) => {
+  return chartItems.slice().sort((a, b) => {
+    if (a.type === "momentPoints") return 1;
+    else if (b.type === "momentPoints") return -1;
+    else if (a.type === "momentLines") return 1;
```

Comment:
Can make the first line `if (a.type === "momentPoints" || a.type === "momentLines") return 1;` instead of having multiple returns  with the same value.

Reply:
It wouldn't correctly sort the values in that case